packages/react/emotion-utils/src/types.ts

```diff
@@ -10,6 +10,9 @@ export type AxisDirection = 'vertical' | 'horizontal';
 export type InferenceHTMLElement<K extends keyof JSX.IntrinsicElements> = NonNullable<
   Extract<JSX.IntrinsicElements[K]['ref'], React.RefObject<any>>['current']
 >;
-export interface AsProps<T extends keyof JSX.IntrinsicElements> extends AllHTMLAttributes<T> {
+
+export type StringElementType = ElementType & string;
+
+export type AsProps<T extends StringElementType> = ComponentPropsWithoutRef<T> & {
```

Comment:
previously, AllHTMLAttributes make component can accept all
but after this Pull Request AsProp will restrict to component accept only as's componentProps